rviz_common/include/rviz_common/ros_integration/ros_node_abstraction.hpp

```diff
@@ -65,19 +63,14 @@ class RosNodeAbstraction : public RosNodeAbstractionIface
   RVIZ_COMMON_PUBLIC
   explicit RosNodeAbstraction(const std::string & node_name);
 
-  /// Creates a ros node with the given name using a specific storage. For testing use only.
+  /// Adds this node to an executor
   /**
-   * Internally a rclcpp::Node is created.
-   * If a rclcpp::Node with the given name already exists a node abstraction for this already
-   * existing node is created.
+   * Adds the rclcpp::Node representation of this to the given executor.
    *
-   * \param node_name name of the node to create
-   * \param ros_node_storage storage handling for the internal rclcpp::Node
+   * \param executor executor to whom the node is added
    */
   RVIZ_COMMON_PUBLIC
-  RosNodeAbstraction(
-    const std::string & node_name,
-    std::shared_ptr<RosNodeStorageIface> ros_node_storage);
+  void add_to_executor(rclcpp::executor::Executor & executor) override;
```

Comment:
This doesn't work, since it breaks the ROS 1/ROS 2 abstraction (the whole point of this class). Instead we'll need to abstract the idea of using an executor. Also the function below that returns the "raw node pointer" also breaks this abstraction.

Reply:
Even if we were to put the raw node pointer function in for now, this one doesn't make sense, because you can just get the node and add it to the executor. There's really no reason that I can see to have this function on the node abstraction class.

Reply:
So far I thought more about refactoring towards easier testability but I guess the goals are very similar.
The accessor the the "raw node" is intended to be removed as soon as the abstraction for subscriptions is ready (which we already started working on). The `add_to_executor` would then be the only way to achieve that. It was also my way of replacing `DisplayContext::addNodeToMainExecutor()`.

Maybe we no longer need this function if we add the rviz ros node to the executor when we initialize the ros client abstraction. Currently the `VisualizationManager` holds the executor.